nexus/db-queries/src/db/datastore/vpc.rs

```diff
@@ -740,6 +753,64 @@ impl DataStore {
             })
     }
 
+    pub async fn nexus_inbound_icmp_view(
+        &self,
+        opctx: &OpContext,
+    ) -> Result<ServiceIcmpConfig, Error> {
+        opctx.authorize(authz::Action::Read, &authz::FLEET).await?;
+        use nexus_db_schema::schema::vpc_firewall_rule::dsl;
+
+        let conn = self.pool_connection_authorized(opctx).await?;
+
+        let rule = dsl::vpc_firewall_rule
+            .filter(dsl::time_deleted.is_null())
+            .filter(dsl::vpc_id.eq(*SERVICES_VPC_ID))
+            .filter(dsl::name.eq(NEXUS_ICMP_FW_RULE_NAME.to_string()))
+            .limit(1)
+            .select(VpcFirewallRule::as_select())
+            .get_result_async::<VpcFirewallRule>(&*conn)
+            .await
+            .map_err(|e| public_error_from_diesel(e, ErrorHandler::Server))?;
```

Comment:
IIUC this will return a 500 error if there is no non-deleted firewall rule with the UUID `SERVICES_VPC_ID` and name `NEXUS_ICMP_FW_RULE_NAME`. I realize that because these are not user-created rules, we expect them to always exist...but I wonder if it would still be more correct to return `NotFound` here if they are not found? 

Or, perhaps we should still return a 500 error in that case, but log a specific warning if the firewall rule isn't in the database, since that would suggest it's been deleted or something. We could use Diesel's [Optional](https://docs.diesel.rs/master/diesel/result/trait.OptionalExtension.html#tymethod.optional) to handle the "not found" case specially and construct our own error with a more specific log message, or something?